### DISCUSSION TITLE
ccls has hardcoded path compiled in binary, bug in formula?

### DISCUSSION THREAD

**Discussion by mrexodia** [Homebrew/discussions#2656] [Writing Formulae/Casks]
The package `ccls` has a hardcoded path the the clang resource directory compiled into the binary. When running it shows the following in the log:

`20:11:30 initialize.cc:329 I use -resource-dir=/usr/local/Cellar/llvm/13.0.0/lib/clang/13.0.0`

But on my system this path is `/usr/local/Cellar/llvm/13.0.0_2/lib/clang/13.0.0`. I have been in discussions about this (https://github.com/MaskRay/ccls/issues/853#issuecomment-997448145) and I have some questions:

1. Is this a bug in the brew formula?
2. Is it possible to force a brew formula to always build from source?

**Comment by gromgit** [comment]
> 1. Is this a bug in the brew formula?

Hard to say. I've followed up in the upstream thread you linked to, but the issue is essentially that the build process configures itself with `clang -print-resource-dir`, which returns the path you mentioned, and is guaranteed to change on even a minor revision (that's the `_2` in the path). I suggested upstream that the path actually be retrieved at run time, so it's always current, but there may be something about `ccls` that won't work with that method.

> 2. Is it possible to force a brew formula to always build from source?

`brew reinstall -s ccls`, but you'd have to run that on every LLVM upgrade.

**Comment by gromgit** [comment]
PR https://github.com/Homebrew/homebrew-core/pull/91706 to fix this has been merged. Please `brew update`, then `brew upgrade ccls`.
